server/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java

```diff
@@ -649,6 +650,9 @@ public static IndexAnalyzers build(
             throw new IllegalArgumentException("no default analyzer configured");
         }
         defaultAnalyzer.checkAllowedInMode(AnalysisMode.ALL);
+        if (Objects.equals(defaultAnalyzer.name(), DEFAULT_ANALYZER_NAME) == false) {
+            throw new IllegalStateException("default analyzer must have the name [default] but was: [" + defaultAnalyzer.name() + "]");
```

Comment:
Can we add a new check for this exception? The existing one was removed with "testAnalyzerMapChecks" I think and I couldn't find a new one.

Reply:
Looking more carefully at how these are constructed, I don't think it's actually possible for the analyzer registered under `default` to *not* be called `default`.  I'll add an assertion, but will keep the exception in place for paranoia's sake.